org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend

```diff
@@ -2390,11 +2447,28 @@ class ResolveTypeHierarchyItemParams {
 
 }
 
+@JsonRpcData
+class DocumentSymbolOptions implements WorkDoneProgressOptions {
+	Boolean workDoneProgress
+}
+
+@JsonRpcData
+class DocumentSymbolRegistrationOptions extends TextDocumentRegistrationOptions implements WorkDoneProgressOptions {
+	Boolean workDoneProgress
+}
+
 /**
  * The document symbol request is sent from the client to the server to list all symbols found in a given text document.
  */
 @JsonRpcData
-class DocumentSymbolParams {
+class DocumentSymbolParams implements WorkDoneProgressParams, PartialResultParams {
+	
+	@JsonAdapter(EitherTypeAdapter.Factory)
```

Comment:
My gut feeling is that we should introduce a base class for all `*Params` that implement `WorkDoneProgressParams, PartialResultParams` and move the fields into that base class. I see that the spec does this differently, but it's only defining the interface and it could be worthwhile to reduce the duplication on the impl side, too.
This would also allow for generic progress implementations without the need to define a signature like
`<T extends WorkDoneProgressParams & PartialResultParams> myMethod(T paramsWithProgress)`

Reply:
ok for me, @spoenemann ?

Reply:
:+1: